include/tvm/ir/module.h

```diff
@@ -349,6 +350,9 @@ class IRModuleNode : public Object {
    */
   std::unordered_set<String> import_set_;
   friend class IRModule;
+
+ public:
+  void ExtractPrimFuncConstants(tir::PrimFunc func);
```

Comment:
I thought we were agreeing to centralize the constants when we add the PrimFunc to the IRModule -- thus PrimFunc could exist before the IRModule.

Moreover, there are scenarios where PrimFuncs are moved to different IRModules (e.g. per-target IRModule). We could keep that movement simpler and update the IRModule when such a move occurs.

Our preference was always has been to stick the constants as PrimFunc/BaseFunc attrs and move them later for IRModule for codegen -- as requested in the RFC discussion. This cross-referencing of constants is to aid the lowering flow to runtime.Modules. 

Imposing a restriction of "we should make sure that the constants are always stored in the IRModule attributes" -- will also create a restriction of PrimFuncs cannot be created without an IRModule.

I ll let @d-smirnov elaborate further on this.

Reply:
Typically PrimFunc are copied in between different IR modules. However `functions` field in IRModule is public which makes extraction of data from PrimFunc to surrounding module quite risky and discipline demanding. It would be better not to store constants outside of PrimFunc at all, or, alternatively to have a global storage for constants available from compilation context.